pkg/agent/agent.go

```diff
@@ -86,6 +86,16 @@ func (k *Klocksmith) process(stop <-chan struct{}) error {
 		return fmt.Errorf("failed to set node info: %v", err)
 	}
 
+	glog.Info("Checking annotations")
+	node, err := k8sutil.GetNodeRetry(k.nc, k.node)
+	if err != nil {
+		return err
+	}
+
+	// Only make a node schedulable if a reboot was in progress. This prevents a node from being made schedulable
+	// if it was made unschedulable by something other thin the agent
+	makeSchedulable := node.Annotations[constants.AnnotationAgentMadeUnschedulable] == constants.True
```

Comment:
When the agent runs for the first time, this annotation doesn't exist, so this value is set as false, and the log message that says "node was marked unschedulable by external source" is printed out, which could be confusing. We should probably use the map multi-return value to check for the key existence and use that in the if statement below.